### PR TITLE
feat: add support for highlight.new

### LINKED ISSUE #1079: Feature: highlight.new path
### Type of feature

🍕 Feature

### Current behavior

We own the [highlight.new](https://highlgiht.new) and got a note from the register. In order to compliant, we linking to the insights page.

## Note from registry

Domain Name | Non Compliant Policy
-- | --
highlight.new | Does not take the user directly into the action generation or content creation flow.

<p style="box-sizing: content-box; letter-spacing: normal; margin: 1em 0px; padding: 0px; border: 0px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; font-size: 14px; line-height: inherit; font-family: -apple-system, &quot;system-ui&quot;, &quot;segoe ui&quot;, roboto, oxygen, ubuntu, cantarell, &quot;fira sans&quot;, &quot;droid sans&quot;, &quot;helvetica neue&quot;, sans-serif; vertical-align: baseline; display: block; color: rgb(0, 0, 0); orphans: 2; text-align: left; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">We would like to clarify that for highlight.new, the policy violation is due to the additional click "Create/Add insight Page" (see image 1) before entering the content creation flow. If the user is brought directly into the creation page screen (see image 2) after login, the website would be compliant.<br style="box-sizing: content-box; letter-spacing: initial;"><br style="box-sizing: content-box; letter-spacing: initial;">Please update your website to be compliant with the above policy.<br style="box-sizing: content-box; letter-spacing: initial;"><br style="box-sizing: content-box; letter-spacing: initial;">If the website is still not compliant in the next 14 days, your domain will be suspended. Suspended domains do not resolve, and cannot be transferred or deleted.<br clear="none" style="box-sizing: content-box; letter-spacing: initial;"><br clear="none" style="box-sizing: content-box; letter-spacing: initial;">Regards,<br clear="none" style="box-sizing: content-box; letter-spacing: initial;">Google Registry</p>Dear Registrant,

Google Registry conducts a weekly analysis of .new domains to verify compliance with our [.new Registration Policies](https://www.registry.google/policies/registration/new/#!/). On April 5, 2023, the following domain that you manage was determined non-compliant with the following policy:

Domain Name 

Non Compliant Policy

highlight.new

Does not take the user directly into the action generation or content creation flow. 

We would like to clarify that for highlight.new, the policy violation is due to the additional click "Create/Add insight Page" (see image 1) before entering the content creation flow. If the user is brought directly into the creation page screen (see image 2) after login, the website would be compliant.

Please update your website to be compliant with the above policy.

If the website is still not compliant in the next 14 days, your domain will be suspended. Suspended domains do not resolve, and cannot be transferred or deleted.

Regards,
Google Registry

### Suggested solution

@getaheaddev need a highlights/new page. 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by diivi**
## Description
Fixes #1079 

added a ?new=true query param that auto clicks and focuses on the highlight creation form so the user can create a highlight.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
#1079 

## Mobile & Desktop Screenshots/Recordings
Peek 2023-08-05 00-46

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
@diivi visiting this link https://deploy-preview-1487--oss-insights.netlify.app/feed?new=true does not force a login currently. It works otherwise

**Comment by diivi** [comment]
Checking if a user is logged in is proving to be harder than I thought :/

Using this code is not working for some reason, I might need more context, as in where I can check this
```
  useEffect(() => {
    const queryParams = new URLSearchParams(window.location.search);
    const newHighlight = queryParams.get("new");

    const checkUserAuth = async () => {
      const sessionData = await loadSession();
      console.log(sessionData)
      // if (newHighlight && !sessionData) {
      //   signIn({ provider: "github", options: { redirectTo: `${window.location.href}` } });
      // }
    };
    void checkUserAuth();

    const focusOnHighlighCreationInput = setInterval(() => {
      const highlightCreationInput = document.getElementById("highlight-create-input");
      if (newHighlight && highlightCreationInput) {
        highlightCreationInput.click();
        highlightCreationInput.focus();
        clearInterval(focusOnHighlighCreationInput);
      }
    }, 1000);

    return () => {
      clearInterval(focusOnHighlighCreationInput);
    };
  }, []);
```

Running this  logs `false` to the console even when I'm logged in.

**Comment by bdougie** [comment]
This logic exists on insights/new

**Comment by diivi** [comment]
> update: Does this help @diivi?
> 
> https://github.com/open-sauced/insights/blob/0cedccc612c5080df3552e9d62580c540a4ac9c9/pages/hub/insights/new.tsx#L11

I checked that, yes.
But this ended up being a little more complex than expected lol. The end solution finally works though.

File: lib/hooks/useSession.ts [hunk 1]
```diff
@@ -53,6 +53,7 @@ const useSession = (getSession = false) => {
     waitlisted,
     hasReports,
     session,
+    loadSession,
```
Comment (diivi) [inline lib/hooks/useSession.ts:56]:
```suggestion
```

**Review by brandonroberts — APPROVED** [review]

**Comment by bdougie** [comment]
Mobile needs the same treatment


 image

File: pages/feed/index.tsx [hunk 2]
```diff
@@ -85,6 +85,29 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
     { name: "Highlights", count: highlights_count ?? 0 },
   ];
 
+  useEffect(() => {
+    const queryParams = new URLSearchParams(window.location.search);
```
Comment (babblebey) [inline pages/feed/index.tsx:89]:
```suggestion
    const { new: newHighlight, signIn: signInRequired } = router.query;
```

Hey @diivi, I'd suggest we destructure the queries from `next/router` here like so..

File: pages/feed/index.tsx [hunk 3]
```diff
@@ -85,6 +85,29 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
...
+    const newHighlight = queryParams.get("new");
```
Comment (babblebey) [inline pages/feed/index.tsx:90]:
```suggestion
```
Removing this line for `router.query`

File: pages/feed/index.tsx [hunk 4]
```diff
@@ -85,6 +85,29 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
...
+    const signInRequired = queryParams.get("signIn");
```
Comment (babblebey) [inline pages/feed/index.tsx:91]:
```suggestion
```
...and this line for `router.query`

File: pages/feed/index.tsx [hunk 5]
```diff
@@ -290,7 +313,10 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
               selectedFilter={selectedRepo}
               setSelected={(repo) => {
                 if (!openSingleHighlight) {
-                  router.push(`/feed${repo ? `?repo=${repo}` : ""}`);
+                  const queryParams = new URLSearchParams(window.location.search);
```
Comment (babblebey) [inline pages/feed/index.tsx:316]: Here too

**Review by babblebey — COMMENTED** [review]
Kindly note, I am just thinking out loud here, if there's a reason why you chose to use the `URLSearchParams` api, I'm curious to know.

**Comment by diivi** [comment]
I'm aware of router.query, but it didn't work when I started development for this PR, so I moved to the browser based solution. I think it was due to race conditions (had some trouble with these on this PR).

If router.query works, I'll use that.

File: pages/feed/index.tsx (hunk 5 above)
Comment (diivi) [reply pages/feed/index.tsx:316]: Didn't get this, can you add a code snippet please?

**Comment by diivi** [comment]
The mobile form should now open up automatically when ?new=true

**Review by bdougie — APPROVED** [review]

File: pages/feed/index.tsx (hunk 5 above)
Comment (babblebey) [reply pages/feed/index.tsx:327]: Here @diivi 

```js
router.push(
  `/feed${repo ? router.asPath `&repo=${repo}` : router.asPath ? router.asPath : ""}`
);
```

File: pages/feed/index.tsx [hunk 6]
```diff
@@ -85,6 +85,40 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
...
 
+  useEffect(() => {
+    const queryParams = new URLSearchParams(window.location.search);
+    const newHighlight = queryParams.get("new");
+    const signInRequired = queryParams.get("signIn");
+
+    if (newHighlight && signInRequired) {
+      signIn({ provider: "github", options: { redirectTo: `${window.location.origin}/feed?new=${newHighlight}` } });
+    }
+    var focusOnHighlighCreationInput: NodeJS.Timeout;
```
Comment (brandonroberts) [inline pages/feed/index.tsx:96]:
```suggestion
    let focusOnHighlighCreationInput: NodeJS.Timeout;
```

File: pages/feed/index.tsx [hunk 7]
```diff
@@ -85,6 +85,40 @@ const Feeds: WithPageLayout<HighlightSSRProps> = (props: HighlightSSRProps) => {
...
+          clearInterval(focusOnHighlighCreationInput);
+        }
+      }, 1000);
+    } else {
+      // for mobile. No need to focus on input, just click on the button as it opens up a form anyway.
+      focusOnHighlighCreationInput = setInterval(() => {
+        const mobileHighlightCreateButton = document.getElementById("mobile-highlight-create-button");
+        if (newHighlight && mobileHighlightCreateButton) {
+          mobileHighlightCreateButton.click();
+          clearInterval(focusOnHighlighCreationInput);
```
Comment (brandonroberts) [inline pages/feed/index.tsx:113]: If the ?new isn't passed on desktop, won't this keep checking forever? Why isn't this a one-time check?

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: pages/feed/index.tsx (hunk 7 above)
Comment (diivi) [reply pages/feed/index.tsx:113]: Yeah I should be creating these intervals only when ?new is passed, I'll change that.

Regarding why this is not a one time check,
I just didn't get one of these elements without waiting for some time (it just doesn't get added to the DOM before this line of code executes).

**Review by brandonroberts — APPROVED** [review]
It works. Question about the delay time

**Comment by brandonroberts** [comment]
@diivi nevermind on the delay. This will continue to check on the interval

**Pull request merged**